File: testing/hash_grouping/benchmark.py
import json, time, argparse
from algo_group import create_schedules

parser = argparse.ArgumentParser()
parser.add_argument('--courses', dest='course_list', help='List of courses to create schedules from',nargs='+')
parser.add_argument('--term', dest='term', help='Term code')
parser.add_argument('--log',dest='log',help='Log results to file',default=False,action='store_true')
args = parser.parse_args()

with open(f'testing/hash_grouping/table-{args.term}.json','r') as file:
    data = json.load(file)

courses = [course for course in data if course['name'] in args.course_list]

start = time.time()
results = create_schedules(courses)
end = time.time()
duration = round(end - start, 2)

print(f'duration: {duration} seconds')
print(f'count: {len(results)}')

if args.log:
    results = [[group['crns'] for group in result] for result in results]
    results.insert(0, {'algorithm': 'hash/group', 'duration': duration + ' seconds', 'count': len(results)})
    with open(f'testing/hash_grouping/results-{args.term}.json','w') as file:
        json.dump(results, file)

File: testing/hash_grouping/js/benchmark.js
const create_schedules = require('./algo_group');
const { program } = require('commander');
const fs = require('fs');

program
    .requiredOption('--term <term>', 'Term to create schedules for')
    .option('--log', 'Log results to a file')
    .requiredOption('--courses <courses...>', 'List of courses to create schedules from')

program.parse();
const options = program.opts();

const data = require(`../table-${options.term}.json`);
const courses = data.filter(c => options.courses.includes(c.name));

const start = process.hrtime();
let results = create_schedules(courses);
const end = process.hrtime(start);
const duration = end[0] + Number((end[1]/1e9).toFixed(2));

console.log(`duration: ${duration} seconds`);
console.log(`count: ${results.length}`);

if (options.log){
    results = results.map(result => result.map(g => g.hash).join(''));
    results.unshift({'algorithm': 'hash/group', 'duration': `${duration} seconds`, 'count': results.length});
    fs.writeFileSync(`results-${options.term}.js.json`, JSON.stringify(results, null, 2));
}